src/main/kotlin/vannak/tech/BookStockManagement/services/BookService.kt

```diff
@@ -0,0 +1,58 @@
+package vannak.tech.BookStockManagement.services
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.data.domain.PageRequest
+import org.springframework.data.domain.Pageable
+import org.springframework.http.ResponseEntity
+import org.springframework.stereotype.Component
+import vannak.tech.BookStockManagement.api.DTOs.CreateBookDTO
+import vannak.tech.BookStockManagement.api.DTOs.UpdateBookDTO
+import vannak.tech.BookStockManagement.domain.models.Book
+import vannak.tech.BookStockManagement.repositories.BookRepository
+import vannak.tech.BookStockManagement.repositories.CategoryRepository
+
+@Component
+class BookService(
+        @Autowired
+        var repository: BookRepository,
+        @Autowired
+        var categoryRepository: CategoryRepository
+) {
+        lateinit var pageable: Pageable
+
+        fun index(categoryId:Long, page:Int, size:Int, q: String?):ResponseEntity<Any>{
```

Comment:
Please refactor this code. no need to filter, in order run repository to get the data